### PR TITLE
fix(ses-ava): wrap test deeply

### PR DESCRIPTION
**Pull request by mhofman**
Refs: https://github.com/Agoric/agoric-sdk/pull/10208

## Description

the `ava` test functions can be nested more than one level deep. This change recurses test functions with known names until none is found.

### Security Considerations

None

### Scaling Considerations

None

### Documentation Considerations

Relaxes the type of acceptable test functions as it only really cares about it being a test function, regardless of whatever chain expandos it may have.

### Testing Considerations

Manually verified that a `test.serial.only` does show errors details as expected. Not sure if we have any automated coverage for this.

### Compatibility Considerations

None

### Upgrade Considerations

This is used by tests only, and can be consumed as a non breaking change simply by using the new NPM package including this fix.

### CODE REVIEW

**Review by kriskowal — APPROVED** [review]
Neat.

File: packages/ses-ava/src/ses-ava-test.js [hunk 1]
```diff
@@ -176,7 +177,7 @@ const makeVirtualExecutionContext = originalT => {
 };
 
 /**
- * @template {import('ava').TestFn} [T=import('ava').TestFn]
+ * @template {import('ava').OnlyFn} [T=import('ava').TestFn]
```
Comment (turadg) [inline packages/ses-ava/src/ses-ava-test.js:180]: @mhofman why this change? it's breaking my Endo sync PR by making `wrapTest` erase the type:

Image

Comment (mhofman) [reply packages/ses-ava/src/ses-ava-test.js:180]: This is 2 years old, and the type was working fine then. I think the reasoning was that this would accept any test fn, including possibly nested ones, not only the top level one that requires more expandos.

Comment (turadg) [reply packages/ses-ava/src/ses-ava-test.js:180]: Yeah, it definitely worked. Maybe something in Ava 7 broke it.
